aws/resource_aws_s3_bucket.go

```diff
@@ -2263,6 +2262,14 @@ func resourceAwsS3BucketLifecycleUpdate(s3conn *s3.S3, d *schema.ResourceData) e
 			}
 		}
 
+		// As a lifecycle rule requires 1 or more transition/expiration actions,
+		// we explicitly pass a default ExpiredObjectDeleteMarker value to be able to create
+		// the rule while keeping the policy unaffected if the conditions are not met.
+		if rule.Expiration == nil && rule.NoncurrentVersionExpiration == nil &&
```

Comment:
workaround for cases like `TestAccAWSS3Bucket_LifecycleRule_Expiration_EmptyConfigurationBlock` ..or is this something we should/can enforce at the schema level instead @bflad ? when `rule["expiration"]` was a `schema.TypeSet` the  `ExpiredObjectDeleteMarker` was set as false in the logic about but this isn't the case when its a list 🤔